layouts/layouts/src/main/java/com/microsoft/device/dualscreen/layouts/FoldableLayoutController.kt

```diff
@@ -0,0 +1,356 @@
+     * @param startScreenRect The bounds for the first screen
+     * @param endScreenRect The bounds for the second screen
+     */
+    private fun addDualScreenContent(
+        layoutOrientation: Int,
+        startScreenRect: Rect,
+        endScreenRect: Rect
+    ) {
+        rootView.orientation = layoutOrientation
+        updateDimensionsForDualScreen(
+            layoutOrientation,
+            startScreenRect,
+            endScreenRect
+        )
+
+        layoutConfig.dualScreenStartLayoutId
+            .takeIf { it != View.NO_ID }?.let { resLayoutId ->
+                firstContainer.addContent(resLayoutId)
+            }
+        layoutConfig.dualScreenEndLayoutId
+            .takeIf { it != View.NO_ID }?.let { resLayoutId ->
+                secondContainer.addContent(resLayoutId)
+            }
+
+        addHingeIfNeeded()
+    }
+
+    /**
+     * Add the hinge view if it's not already added
+     */
+    private fun addHingeIfNeeded() {
+        if (!rootView.hasChild(R.id.hinge_id)) {
+            foldingFeature?.generateHingeView(rootView.context)?.let { hingeView ->
+                rootView.addView(hingeView, 1)
+            }
+        }
+    }
+
+    /**
+     * Removes the hinge view from [FoldableLayout] root view.
+     */
+    private fun removeHinge() {
+        rootView.findViewById<View>(R.id.hinge_id)?.let {
+            rootView.removeView(it)
+        }
+    }
+
+    /**
+     * Updates the containers dimensions corresponding to the single screen mode
+     */
+    private fun updateDimensionsForSingleScreen() {
+        firstContainer.updateLayoutParams<LinearLayout.LayoutParams> {
+            width = MATCH_PARENT
+            height = MATCH_PARENT
+        }
+        secondContainer.updateLayoutParams<LinearLayout.LayoutParams> {
+            width = WRAP_CONTENT
+            height = MATCH_PARENT
+        }
+    }
+
+    /**
+     * Updates the containers dimensions corresponding to the dual screen mode and layout orientation
+     *
+     * @param linearLayoutOrientation The [FoldableLayout] orientation
+     * @param startScreenRect The bounds for the first screen
+     * @param endScreenRect The bounds for the second screen
+     */
+    private fun updateDimensionsForDualScreen(
+        linearLayoutOrientation: Int,
+        startScreenRect: Rect,
+        endScreenRect: Rect
+    ) {
+        if (linearLayoutOrientation == VERTICAL) {
+            resetContainersDimensions()
+
+            rootView.doOnNextLayout {
+                updateDimensionsForDualLandscape()
+            }
+        } else {
+            updateDimensionsForDualPortrait(startScreenRect, endScreenRect)
+        }
+    }
+
+    /**
+     * Updates the containers dimensions corresponding to the dual screen mode in portrait orientation
+     *
+     * @param startScreenRect The bounds for the first screen
+     * @param endScreenRect The bounds for the second screen
```

Comment:
nit, but it would be more correct to say: first display area and second display area since not all devices have two physical screens.

Reply:
Done